htlcswitch.go

```diff
@@ -443,3 +502,36 @@ type linkInfoUpdateMsg struct {
 func (h *htlcSwitch) UpdateLink(chanPoint *wire.OutPoint, bandwidthDelta btcutil.Amount) {
 	h.linkControl <- &linkInfoUpdateMsg{chanPoint, bandwidthDelta}
 }
+
+// Message to ask if it is possible to send given amount of money to a partner.
```

Comment:
I don't think this message along with the functions to handle it are necessary. Instead an attempt should be made to send an HTLC by the switch in the loop above (as it is currently), with an error being returned to the user if the current available link capacity is insufficient.